src/psycopt2d/visualization/sens_over_time.py

```diff
@@ -93,15 +95,114 @@ def create_sensitivity_by_time_to_outcome_df(
     return output_df
 
 
+def _generate_sensitivity_array(
+    df: pd.DataFrame,
+    y_n_decimals: int,
+) -> np.ndarray:
+    """Generate sensitivity array for plotting heatmap.
+
+    Args:
+        df (pd.DataFrame): Dataframe with columns "sens", "days_to_outcome_binned" and "threshold".
+
+    Returns:
+        np.ndarray
+    """
+    x_labels = df["days_to_outcome_binned"].unique().tolist()
+    y_labels = df["threshold"].unique().tolist()
+
+    sens_array = []
+    for threshold in y_labels:
+        sens_for_threshold = []
+        small_df = df[df.threshold == threshold]
+        for days_interval in x_labels:
+            smaller_df = small_df[small_df["days_to_outcome_binned"] == days_interval]
+            if len(smaller_df["sens"].unique().tolist()) > 1:
+                raise ValueError(
+                    "More than one sensitivity value for this threshold and days interval",
+                )
+            else:
+                sens_for_threshold.append(smaller_df["sens"].unique().tolist()[0])
+
+        sens_array.append(sens_for_threshold)
+
+    return (
+        np.array(sens_array),
+        x_labels,
+        [round(y_labels[value], y_n_decimals) for value in range(len(y_labels))],
+    )
+
+
+def _annotate_heatmap(
+    im: matplotlib.image.AxesImage,
+    data: Optional[np.array] = None,
+    valfmt: Optional[str] = "{x:.2f}",
```

Comment:
```suggestion
    value_formatter: Optional[str] = "{x:.2f}",
```
Might want to change name with f2